src/core_modules/capture-core/components/WidgetEventEdit/EditEventDataEntry/editEventDataEntry.epics.js

```diff
@@ -45,37 +45,36 @@ const getDataEntryId = (event): string => (
         : dataEntryIds.SINGLE_EVENT
 );
 
-export const loadEditEventDataEntryEpic = (action$: InputObservable, store: ReduxStore) =>
+export const loadEditEventDataEntryEpic = (action$: InputObservable, store: ReduxStore, { querySingleResource }: ApiUtils) =>
     action$.pipe(
         ofType(eventDetailsActionTypes.START_SHOW_EDIT_EVENT_DATA_ENTRY, widgetEventEditActionTypes.START_SHOW_EDIT_EVENT_DATA_ENTRY),
-        map((action) => {
+        switchMap((action) => {
             const state = store.value;
             const loadedValues = state.viewEventPage.loadedValues;
             const eventContainer = loadedValues.eventContainer;
             const metadataContainer = getProgramAndStageFromEvent(eventContainer.event);
             if (metadataContainer.error) {
-                return prerequisitesErrorLoadingEditEventDataEntry(metadataContainer.error);
+                prerequisitesErrorLoadingEditEventDataEntry(metadataContainer.error);
+                return EMPTY;
```

Comment:
Why not use `return from(prerequisitesErrorLoadingEditEventDataEntry(metadataContainer.error))`?

Reply:
When using `from()`, the code throws a runtime error. This happens because `prerequisitesErrorLoadingEditEventDataEntry` returns a string and it is not what `from()` expects. Instead, I use `of()` to create an Observable, which works fine for this case.